docs/user/advanced_interfaces.rst

```diff
@@ -0,0 +1,146 @@
+and re-build the image:
+
+#. On the project's landing page, click the **View in Gitlab** button in the upper righthand corner (opens a new tab by default).
+#. In Gitlab, click the **Repository** tab in the lefthand column, which drops you into the **Files** tab.
+#. Click the **Dockerfile** out of the list of files that appears, and click **Edit** (top right, near the red Delete button. Don't click the red Delete button.)
+#. Make your edits in this window.
+#. When you're satisfied with the edits, scroll down and write a meaningful **commit message** (you'll thank yourself later).
+#. Click the green **Commit changes** button.
+
+Now you've committed the changes to your Dockerfile. Since you've made a commit
+and haven't modified any of the gitlab CI settings, the CI/CD pipeline will kick
+off. It will attempt to rebuild your project with the new contents of your
+Dockerfile based on the configuration in ``.gitlab-ci.yml``, a file at the top
+level of your project directory.
+
+The contents of ``.gitlab-ci.yml`` show you that in the build stage, we pull
+the docker image for Renku, build our new image out of our Dockerfile
+with a tag relating to the commit, and push it.
+
+Let's monitor this process:
+
+#. Click the **CI/CD** > **Jobs** tab.
+#. Click the latest status that corresponds to the changes to the Dockerfile you just made (probably "running", unless it's already "completed" or "failed").
+
+This is the log file from the build process specified in the ``.gitlab-ci.yml``
+file. If it succeeds, there will be a green **passed** status, and the end of
+log will be a green **Job succeeded**. If the build instead failed, you can use
+the messages in the log to determine why and hint at what you can do to fix it.
+
+.. warning::
+
+  Note that the settings have been configured for this build to time out and fail
+  after one hour. While a long running build might be indicative of a bug in your
+  Dockerfile, it's possible that your build might take a long time. If this is the
+  case, you can change these settings via the lefthand column of the Gitlab
+  interface at **Settings** > **CI/CD** > **General pipelines** > **Timeout**.
+
+Using your new Dockerfile
+"""""""""""""""""""""""""
+
+Passing CI/CD is great, but in order to use the new image you need to
+(re)start your interactive environment.
+
+To do this, go back to the Renku platform (hopefully you didn't close the tab),
+and from the project's landing page, first check in the **Files** tab that your
+changes to the Dockerfile are present. If not, you can force-refresh the page.
+Then, go to the **Notebook servers** tab, and kill any running notebooks. Once
+those are shut down, you can **Start new server** and **Launch server**.
+
+If the server launches, test it to make sure that the extra functionality you
+added in the Dockerfile is present in the container. If it isn't, you can go back
+to the Gitlab interface and continue to make changes until you are satisfied.
```

Comment:
done for the docs that I own -- there were some other docs in other sections that don't match this pattern. I didn't change them because I wasn't sure if some of them mattered.

Reply:
great, thanks - we should standardize this as much as possible